src/scrot.c

```diff
@@ -335,38 +335,67 @@ void scrotDoDelay(void)
         dprintf(STDERR_FILENO, "Taking shot in ");
         for (int i = opt.delay; i > 0; i--) {
             dprintf(STDERR_FILENO, "%d.. ", i);
-            opt.delayStart.tv_sec++;
-            scrotWaitUntil(&opt.delayStart);
+            opt.delayStart = scrotSleepFor(opt.delayStart, 1000);
         }
         dprintf(STDERR_FILENO, "0.\n");
     } else {
-        opt.delayStart.tv_sec += opt.delay;
-        scrotWaitUntil(&opt.delayStart);
+        scrotSleepFor(opt.delayStart, opt.delay * 1000);
     }
 }
 
-/* scrotWaitUntil: clock_nanosleep with a simpler interface and no EINTR nagging
+static long miliToNanoSec(int ms)
+{
+    return ms * 1000L * 1000L;
+}
+
+/* clockNow() has the exact same semantics as CLOCK_MONOTONIC. Except that on
+ * Linux, CLOCK_MONOTONIC does not progress while the system is suspended, so
+ * the non-standard CLOCK_BOOTTIME is used instead to avoid this bug.
  */
-static void scrotWaitUntil(const struct timespec *time)
+struct timespec clockNow(void)
 {
-    /* OpenBSD and OS X lack clock_nanosleep(), so we use a trivial algorithm to
-     * correct for drift and call nanosleep() everywhere.
-     */
+    struct timespec ret;
+#if defined(__linux__)
```

Comment:
Didn't think the `CONTINUOUS_CLOCK` needed to be "public" anymore - since you can just use `clockNow()` instead. So the `ifdef` can be done internally for `clockNow`.

(Though the main reason I added this function was to work around the awkwardness of `clock_gettime`'s outparam, e.g `scrotSleepFor(clockNow(), 64)` is much easier to use.)